### PR TITLE
Fix schema display on http api

### LINKED ISSUE #4753: bug: schema column empty on "SHOW DATABASES" statement in the HTTP query api
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

latest

### What's Wrong?

origin_img_v2_494a087a-ad5c-4634-b2b0-9532e748991g

but it's ok to SELECT system.tables

### What You Expected?

it should display the column fields in the SHOW DATABASES statement.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by flaneur2020**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

currently the schema is taken from plan.schema:

Screen Shot 2022-04-08 at 9 53 42 AM

but in the "SHOW xxx" statements, plan is tranformed into the SELECT query, so the plan struct can not really get the actual schema of the statement.

it might be more accurate to get the schema from the DataBlock, as it always binds the schema with the data inside it.

changes:

- wraps JsonBlock with a schema reference, which inherits from the original DataBlock.
- make JsonBlock owns by Page instead of Arc
- rename ExecuteStateName to ExecuteStateKind
- rename block_to_json.rs to json_block.rs

## Changelog

- Bug Fix

## Related Issues

Fixes #4753

### CODE REVIEW

File: query/tests/it/servers/http/http_query_handlers.rs [hunk 1]
```diff
@@ -128,7 +136,7 @@ async fn test_async() -> Result<()> {
         assert!(result.next_uri.is_none());
         assert!(result.schema.is_none());
```
Comment (youngsofun) [inline query/tests/it/servers/http/http_query_handlers.rs:137]: > assert!(result.schema.is_none()); 

change is_none to is_some()

Comment (youngsofun) [reply query/tests/it/servers/http/http_query_handlers.rs:137]: before this change, only the first response contains schema.

now every response may contain or not contain schema,  only depends on if data is empty.

**Comment by youngsofun** [comment]
only one question:

it may take some time to wait for the first block to be generated.
may be not what user (like UI @Chasen-Zhang ) expected?

File: query/tests/it/servers/http/http_query_handlers.rs (hunk 1 above)
Comment (flaneur2020) [reply query/tests/it/servers/http/http_query_handlers.rs]: fixed, thank you for remind

**Comment by flaneur2020** [comment]
> only one question:
> 
> it may take some time to wait for the first block to be generated. may be not what user (like UI @Chasen-Zhang ) expected?

i guess it's ok to get the data & schema together but later, a schema without data maybe not very useful for the UI, in the most time, the UI would display a progressing view until the first block arrives, then it displays the table with data & schema together 🤔

**Pull request merged**